### PR TITLE
Go SDK Katas: Update beam module dependency

### LINKED ISSUE #22751: [Task]: Update Go SDK Code Katas Beam module dependency
### What needs to happen?

The Beam module dependency listed at [learning/katas/go/go.mod](https://github.com/apache/beam/blob/master/learning/katas/go/go.mod#L22) needs to change from:

```
github.com/apache/beam v2.23.0+incompatible
```

to:

```
github.com/apache/beam/sdks/v2/go/pkg/beam
```

### Issue Priority

Priority: 3

### Issue Component

Component: katas

### PR DESCRIPTION
**Pull request by damondouglas**
This PR fixes #22751 by updating the Beam SDK learning katas module dependency.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - ~Update `CHANGES.md` with noteworthy changes.~
 - ~If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).~

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by damondouglas** [comment]
R: @lostluck

File: build.gradle.kts [hunk 1]
```diff
@@ -100,6 +100,8 @@ tasks.rat {
     "**/*.json",
 
     // Katas files
+    "learning/katas/**/course-info.yaml",
+    "learning/katas/**/task-info.yaml",
```
Comment (damondouglas) [inline build.gradle.kts:104]: IntelliJ edu course creation tool updates these files and removes the license.  Whenever one updates anything in the course files, they would have to manually update the license information.  These files only store metadata about the course.

**Review by lostluck — APPROVED** [review]
LGTM Thanks for the update! Fixing the imports to reflect the production usage is most excellent.

The SDK has triggers now, and in 2.42.0 it looks like we'll have State, if not timers, so the course can expand!  But we most certainly want to wait for a slightly easier to use, portable Go runner before having a stand alone kata course demo those features.

File: build.gradle.kts (hunk 1 above)
Comment (lostluck) [reply build.gradle.kts:104]: Makes sense! Lets not fight IntelliJ here, it's not a good use of anyone's time.

**Comment by damondouglas** [comment]
@lostluck Thank you so much for reviewing this!  We are in the process of building [Tour of Beam](https://github.com/apache/beam/pull/22794).  So perhaps we can add course content there.

**Pull request merged**